src/ServiceBouncer/MainForm.cs

```diff
@@ -137,20 +141,62 @@ private void InstallClicked(object sender, EventArgs e)
 
         private async void Reload()
         {
-            var systemServices = await Task.Run(() => ServiceController.GetServices().Where(service => service.DisplayName.IndexOf(toolStripFilterBox.Text, StringComparison.OrdinalIgnoreCase) >= 0));
-            services.Clear();
-            foreach (var model in systemServices.Select(service => new ServiceViewModel(service)).OrderBy(x => x.Name))
+            try
+            {
+                var systemServices = await Task.Run(() => ServiceController.GetServices(MachineHostname).Where(service => service.DisplayName.IndexOf(toolStripFilterBox.Text, StringComparison.OrdinalIgnoreCase) >= 0));
```

Comment:
How does this handle authentication if your current running user isn't a domain user, or the machine being connected to is on a different domain.

Could we add a user/password entry option for this case?

Reply:
Currently it's all based on the running user. I could change it to be able to provide credentials though. I'm wondering if it might just be the same to use RunAs on the program though?

Reply:
I think that adding ability to run connect as a different user will mean someone doesn't have to keep disconnecting should they only have local users and not a domain user to connect as.

Reply:
Logic like this will allow for impresonation: https://stackoverflow.com/a/212673
We could prompt user to ask if a different login user is required, and if so, show a login form which has the information pushed into the impersonate?